File: src/main/java/org/kirhgoff/munchkin/UserCommandsReader.java
package org.kirhgoff.munchkin;

import java.io.OutputStream;
import java.io.PrintWriter;
import java.util.Scanner;

public class UserCommandsReader implements Runnable{
  private final Scanner scanner;
  private final OutputStream outputStream;
  private final PrintWriter writer;

  public UserCommandsReader (OutputStream outputStream) {
    this.outputStream = outputStream;
    this.writer = new PrintWriter(outputStream);
    this.scanner = new Scanner(System.in);
  }

  public void run() {
    boolean keepRunning = true;

    while(keepRunning) {
      String command = scanner.next();
      if (command.equals("quitnow")) {
        keepRunning = false;
      }
      System.out.println("user>" + command);
      writer.println(command);
      writer.flush();
    }

    System.out.println("System> Quit reader");
  }
}

File: src/main/scala/org/kirhgoff/munchkin/UserCommandsReader.scala
package org.kirhgoff.munchkin

import java.io.PrintWriter
import java.util.Scanner
import scala.util.control.Breaks._

class UserCommandsReader(val output:PrintWriter) extends Runnable{
  val scanner = new Scanner(System.in)
  override def run() = {
    while(true) {
      val command = scanner.next()
      println(s"user>$command")
      output.println(command)
      output.flush()
      if (command.equals("quitnow")) break
    }
    println("system> Quit reader")
  }
}
